modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/transfer/ProcessImport.java

```diff
@@ -0,0 +1,554 @@
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.w3c.dom.Document;
+import org.wso2.carbon.governance.api.util.GovernanceUtils;
+import org.wso2.carbon.pc.core.ProcessCenterConstants;
+import org.wso2.carbon.pc.core.ProcessCenterException;
+import org.wso2.carbon.pc.core.audit.util.RegPermissionUtil;
+import org.wso2.carbon.pc.core.internal.ProcessCenterServerHolder;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.service.RegistryService;
+import org.wso2.carbon.registry.core.session.UserRegistry;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.xml.sax.SAXException;
+
+import javax.ws.rs.core.MediaType;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.wso2.carbon.pc.core.ProcessStore;
+
+public class ProcessImport {
+    private static final Log log = LogFactory.getLog(ProcessImport.class);
+    File[] listOfProcessDirs;
+    RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+    UserRegistry reg;
+    String user;
+
+    /**
+     * @param processZipInputStream
+     * @param user
+     * @throws IOException
+     * @throws RegistryException
+     * @throws ProcessCenterException
+     * @throws UserStoreException
+     */
+    public void importProcesses(InputStream processZipInputStream, String user)
+            throws IOException, RegistryException, ProcessCenterException, UserStoreException, JSONException,
+            TransformerException, SAXException, ParserConfigurationException {
+
+        if (registryService != null) {
+            reg = registryService.getGovernanceUserRegistry(user);
+            this.user = user;
+            //extract zip file stream to the system disk
+            byte[] buffer = new byte[2048];
+            ZipInputStream zipInputStream = new ZipInputStream(processZipInputStream);
+            File importsDir = new File(ProcessCenterConstants.IMPORTS_DIR);
+            if (importsDir != null) {
+                importsDir.mkdirs();
+
+                try {
+                    ZipEntry entry;
+                    while ((entry = zipInputStream.getNextEntry()) != null) {
+                        //counter++;
+                        String outpath = ProcessCenterConstants.IMPORTS_DIR + "/" + entry.getName();
+                        String dirPath = outpath.substring(0, outpath.lastIndexOf("/"));
```

Comment:
Use **File.separator** instead of "/" every place possible. Because Linux uses forward slash and Windows uses back slash.